compiler/rustc_parse/src/parser/path.rs

```diff
@@ -534,4 +537,51 @@ impl<'a> Parser<'a> {
         };
         Ok(Some(arg))
     }
+
+    fn get_ident_from_generic_arg(
+        &self,
+        gen_arg: Option<GenericArg>,
+        lo: Span,
+    ) -> PResult<'a, (Ident, Option<GenericArgs>)> {
+        if let Some(ref arg) = gen_arg {
```

Comment:
Why does not having `gen_arg` result in an error?
Not sure how this works, `Trait<A = B>` is legal code on stable.

Reply:
`parse_generic_arg` would return `GenericArg::Type` containing the `Ident` `A` here. We check for three cases inside `parse_generic_arg`, whether the next token is a lifetime, a const_arg or a type. None is returned in the else clause, I'm not sure what falls under this case and whether we should catch this in an Error if there is a `Colon` or `Eq` token following. I think Trait< : SomeBound> would result in gen_arg being None, in which case we should probably throw an Error indicating that an Identifier has to precede `Colon`.